wetterdienst/provider/noaa/ghcn/api.py

```diff
@@ -0,0 +1,171 @@
+    )
+
+    def _tidy_up_df(self, df: pd.DataFrame, dataset) -> pd.DataFrame:
+        pass
+
+    def _collect_station_parameter(
+        self, station_id: str, parameter, dataset
+    ) -> pd.DataFrame:
+        """
+        Collection method for NOAA GHCN data. Parameter and dataset can be ignored as data
+        is provided as a whole.
+
+        :param station_id: station id of the station being queried
+        :param parameter: parameter being queried
+        :param dataset: dataset being queried
+        :return: dataframe with read data
+        """
+        # parameter and dataset can be ignored as data is provided as a whole
+
+        url = self._base_url.format(station_id=station_id)
+
+        r = requests.get(url)
+
+        r.raise_for_status()
+
+        file = BytesIO(r.content)
+
+        columns = (
+            Columns.STATION_ID.value,
+            Columns.DATE.value,
+            Columns.PARAMETER.value,
+            Columns.VALUE.value,
+            Columns.QUALITY.value,
+        )
+
+        df = pd.read_csv(file, sep=",", compression="gzip", header=None)
+
+        df = df.iloc[:, :-3]
+
+        df.columns = columns
+
+        df.loc[:, Columns.PARAMETER.value] = df.loc[
+            :, Columns.PARAMETER.value
+        ].str.lower()
+
+        return df
+
+
+class NoaaGhcnRequest(ScalarRequestCore):
+    _dataset_base = NoaaGhcnDatasetBase
+    _parameter_base = NoaaGhcnParameter
+    _dataset_tree = NoaaGhcnParameter
+
+    _base_url = "https://www1.ncdc.noaa.gov/pub/data/ghcn/daily/ghcnd-stations.txt"
+
+    _data_range = DataRange.FIXED
+
+    _has_datasets = True
+    _unique_dataset = True
+
+    _period_type = PeriodType.FIXED
+    _period_base = Period.HISTORICAL
+
+    _resolution_type = ResolutionType.FIXED
+    _resolution_base = NoaaGhcnResolution
+
+    provider = Provider.NOAA
+    kind = Kind.OBSERVATION
+
+    _unit_tree = NoaaGhcnUnit
+
+    _values = NoaaGhcnValues
+
+    _tz = Timezone.USA
+
+    def __init__(
+        self,
+        parameter: List[str],
+        start_date: Optional[Union[str, datetime, pd.Timestamp]] = None,
+        end_date: Optional[Union[str, datetime, pd.Timestamp]] = None,
+        humanize: bool = True,
+        tidy: bool = True,
+        si_units: bool = True,
+    ):
+        super(NoaaGhcnRequest, self).__init__(
+            parameter=parameter,
+            resolution=Resolution.DAILY,
+            period=Period.HISTORICAL,
+            start_date=start_date,
+            end_date=end_date,
+            humanize=humanize,
+            tidy=tidy,
+            si_units=si_units,
+        )
+
+    def _all(self) -> pd.DataFrame:
+
+        r = requests.get(self._base_url)
+
+        r.raise_for_status()
+
+        df = pd.read_fwf(
+            BytesIO(r.content),
+            dtype=str,
+            header=None,
+            colspecs=[(0, 11), (12, 20), (21, 30), (31, 37), (41, 71)],
```

Comment:
This looks dangerous to me. Is it possible to add a tiny test, to check out if the format persist? E.g. check out the first line maybe?

Reply:
I carefully selected those values based on the file provided by GHCN at https://www1.ncdc.noaa.gov/pub/data/ghcn/daily/ghcnd-stations.txt so I think this is absolutely fine as there are no further individual listings.